File: article-test/app/src/main/java/com/test/article/ArticleViewModelFactory.java
package com.test.article;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class ArticleViewModelFactory implements ViewModelProvider.Factory {
    private Application application;
    private int id;


    public ArticleViewModelFactory(Application application, int id) {
        this.application = application;
        this.id = id;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new ArticleDetailViewModel(application, id);
    }
}

File: article-test/app/src/main/java/com/test/article/ArticleViewModelFactory.kt
package com.test.article

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class ArticleViewModelFactory(private val application: Application, private val id: Int) :
    ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return ArticleDetailViewModel(application, id) as T
    }

}
